src/Controller/MainController.php

```diff
@@ -39,8 +39,17 @@ public function indexAction(Request $request)
     {
         $response = new Response();
         $threads = $this->threadRepository->findAll();
```

Comment:
Running a `findAll` is a bit dangerous here. If there are several million posts then the DB will send them all back. It would be better to ask the DB how many records there are and then just return the section you need. Look into `count`, `limit` and `offset` functions for the repository

Reply:
yes that way you could pass some limits to the repository and your only querying the database for the results you need. You can do something like this:
```
$threads = $this->threadRepository
            ->createQueryBuilder('t')
            ->setFirstResult(0)
            ->setMaxResults(10)
            ->getQuery()
            ->execute()
;
```
Then set First result to whatever it needs to be for your page